src/index.js

```diff
@@ -79,9 +83,42 @@ export default class MarkersPlugin extends UICorePlugin {
       internalMarker.emitter.off("tooltipChanged", internalMarker.tooltipChangedHandler)
     }
     internalMarker.onDestroy()
+    this._markers.splice(index, 1)
     return true
   }
 
+  /*
+   * Clear all existing markers
+   */
+  clearMarkers() {
+    let self = this
+    if (!this._markers) {
+      return
+    } 
+    // Create a copy of the array since you cannot iterate and remove all its members correctly at the same time
+    let markersCopy = this._markers.slice(0)
+    markersCopy.forEach((marker, index) => {
+      self._removeInternalMarker(marker, index)
+    })
+  }
+
+  /*
+   * Get all markers
+   */
+  getAll() {
```

Comment:
Fair enough. One thing I noticed is that markerLeft property sometimes has NaN as the value. Needs checking.